auth0/src/main/java/com/auth0/android/request/internal/JwksDeserializer.java

```diff
@@ -0,0 +1,66 @@
+package com.auth0.android.request.internal;
+
+import android.util.Base64;
+import android.util.Log;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+class JwksDeserializer implements JsonDeserializer<Map<String, PublicKey>> {
+
+    private static final String RSA_ALGORITHM = "RS256";
+    private static final String USE_SIGNING = "sig";
+
+    @Override
+    public Map<String, PublicKey> deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+        if (!json.isJsonObject() || json.isJsonNull() || json.getAsJsonObject().entrySet().isEmpty()) {
+            throw new JsonParseException("jwks json is not a valid json object");
+        }
+
+        HashMap<String, PublicKey> jwks = new HashMap<>();
+
+        JsonObject object = json.getAsJsonObject();
+        JsonArray keys = object.getAsJsonArray("keys");
+        for (JsonElement k : keys) {
+            JsonObject currentKey = k.getAsJsonObject();
+            String keyAlg = context.deserialize(currentKey.remove("alg"), String.class);
```

Comment:
Why use `remove` which mutates the object, instead of `get`? This question also applies to other usages of `remove` below.

Reply:
Good catch. On other scenarios we would end up keeping the remaining claims as an "extra props" param. We don't do this here so this can be changed to a regular `get(key)`.